url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_ScatterPlotAndLinearRegression.java

```diff
@@ -0,0 +1,902 @@
+//                text_String = "0";
+//            } else {
+//                //text_String = "" + y.stripTrailingZeros().toPlainString();
+//                //text_String = "" + y.round(mc).stripTrailingZeros().toString();
+//                text_String = "" + Generic_BigDecimal.roundStrippingTrailingZeros(
+//                        y,
+//                        Generic_BigDecimal.getDecimalPlacePrecision(y, significantDigits),
+//                        _RoundingMode).toString();
+//            }
+//            text_Width = getTextWidth(text_String);
+//            yAxisMaxLabelWidth = Math.max(yAxisMaxLabelWidth, text_Width);
+//            drawString(
+//                    text_String,
+//                    col - scaleTickAndTextSeparation - scaleTickLength - text_Width,
+//                    row + (textHeight / 3));
+//        }
+//        yAxisExtraWidthLeft += scaleTickLength + scaleTickAndTextSeparation + yAxisMaxLabelWidth;
+//
+//        // Add the Y axis label
+//        setPaint(Color.BLACK);
+//        text_String = yAxisLabel;
+//        text_Width = getTextWidth(text_String);
+//        yAxisExtraWidthLeft += (textHeight * 2) + partTitleGap;
+//        double angle = 3.0d * Math.PI / 2.0d;
+//        writeText(
+//                text_String,
+//                angle,
+//                3 * textHeight / 2,
+//                dataMiddleRow + (text_Width / 2));
+//        // Draw Y axis
+//        setPaint(Color.LIGHT_GRAY);
+//        ab = new Line2D.Double(
+//                originCol,
+//                dataStartRow,
+//                originCol,
+//                dataEndRow);
+//
+//        draw(ab);
+//
+//
+//        result[0] = yAxisExtraWidthLeft;
+//        return result;
+//
+//    }
+//
+//    @Override
+//    public void setOriginCol() {
+//        originCol = coordinateToScreenCol(BigDecimal.ZERO);
+//        System.out.println("originCol " + originCol);
+////        if (minX.compareTo(BigDecimal.ZERO) == 0) {
+////            originCol = dataStartCol;
+////            //originCol = dataStartCol - dataEndCol / 2;
+////        } else {
+////            if (cellWidth.compareTo(BigDecimal.ZERO) == 0) {
+////                originCol = dataStartCol;
+////            } else {
+////                originCol = Generic_BigDecimal.divideRoundIfNecessary(
+////                        minX,
+////                        cellWidth,
+////                        0,
+////                        _RoundingMode).intValueExact()
+////                        + dataStartCol;
+////            }
+////        }
+//    }
+    /**
+     * @param regressionParameters regressionParameters[0] is the y axis
+     * intercept; regressionParameters[1] is the change in y relative to x
+     * (gradient or slope); regressionParameters[2] is the rank correlation
+     * coefficient (RSquare); regressionParameters[3] is data[0].length.
+     */
+    protected void drawLegend(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 92 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 571-651 into a separate method.